org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/views/MavenLifecycleMappingsView.java

```diff
@@ -75,6 +65,10 @@ private void handleSelectionChanged(ISelection selection) {
     Object element;
     if(selection instanceof IStructuredSelection structuredSelection) {
       element = structuredSelection.getFirstElement();
+    } else if(selection instanceof ITextSelection) { // A P
+      element = Optional.ofNullable(PlatformUI.getWorkbench().getActiveWorkbenchWindow())
+          .map(IWorkbenchWindow::getActivePage).map(IWorkbenchPage::getActiveEditor).map(IEditorPart::getEditorInput)
```

Comment:
I'm in favor of checking in JDT formatter settings to "never join wrapped lines". That would allow writing it as
```
stream
.map
.map
.orElse
```
Opinions?

Reply:
Personally I don't think it would always be good. Sometimes it is but sometimes is actually better if the lines are joined. And I think it would be bad if one has to join them manually.
What I usually do if I want such multi-line stream-pipelines is to write a (usually empty) comment behind the last segment to prevent the formatter from joining the line.
That is indeed not idea, but never joining lines isn't IMHO either.

Reply:
I must confess that I don't get what the editor has to do with that. Also should not the view already has a reference to the Workbench/Window?

Reply:
> Also should not the view already has a reference to the Workbench/Window?

Should be done now.